control/state.py

```diff
@@ -163,156 +272,214 @@ def _remove_key(self, key: str):
             self.logger.error(f"Unable to remove key from omap: {ex}. Exiting!")
             raise
 
-    def add_bdev(self, bdev_name: str, val: str):
-        """Adds a bdev to the OMAP."""
-        key = self.BDEV_PREFIX + bdev_name
-        self._add_key(key, val)
+        if self.use_notify:  # Notify other gateways within the group of change
+            try:
+                notify = self.ioctx.notify(self.omap_name)
+            except Exception as ex:
+                self.logger.info(f"Failed to notify.")
 
-    def remove_bdev(self, bdev_name: str):
-        """Removes a bdev from the OMAP."""
-        key = self.BDEV_PREFIX + bdev_name
-        self._remove_key(key)
+    def delete_state(self):
+        """Deletes OMAP object contents."""
+        try:
+            with rados.WriteOpCtx() as write_op:
+                self.ioctx.clear_omap(write_op)
+                self.ioctx.operate_write_op(write_op, self.omap_name)
+                self.ioctx.set_omap(write_op, (self.OMAP_VERSION_KEY,),
+                                    (str(1),))
+                self.ioctx.operate_write_op(write_op, self.omap_name)
+                self.logger.info(f"Deleted OMAP contents.")
+        except Exception as ex:
+            self.logger.error(f"Error deleting OMAP contents: {ex}. Exiting!")
+            raise
 
-    def _restore_bdevs(self, omap_dict, callback):
-        """Restores a bdev from the OMAP."""
+    def update(self, callback):
```

Comment:
Why not start watch/notify via `__init__` or in a different function `register_watch` or similar ? `update` doesn't seem the right place to do this. The semantic for the other update state calls (other classes) is immediate but update here it is enabling some callback mechanism.